File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/UpdateScmActionFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import com.google.inject.Inject;
import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.UpdateScmAction;
import org.eazyportal.plugin.release.core.scm.ScmActions;
import org.eazyportal.plugin.release.jenkins.ReleaseStepConfig;

import java.io.Serializable;

@Extension
public class UpdateScmActionFactory extends InvisibleAction implements Serializable {

    @Inject
    private transient ReleaseStepConfig releaseStepConfig;

    public UpdateScmAction create(ScmActions scmActions) {
        return new UpdateScmAction(
            scmActions,
            releaseStepConfig.getScmConfig()
        );
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/UpdateScmActionFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.UpdateScmAction
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension

class UpdateScmActionFactory {

    fun create(extension: EazyReleasePluginExtension): UpdateScmAction =
        UpdateScmAction(
            extension.scmActions,
            extension.scmConfig
        )

}